### DISCUSSION TITLE
Chrono Error

### DISCUSSION THREAD

**Discussion by dorton21** [idaholab/moose#16599] [Q&A Getting Started] [answered]
Perhaps I'm missing something obvious but while compiling the Makefile within the test folder, I get this error.
For further context I am using GCC version 8.4.0 and openmpi as the hoc wrapper. I played with editing the PerfNode.h file by adding in a "#include <chrono>" statement, however, unless this is a bug, I do not think this should be the solution. I would appreciate any suggestions anyone may have.

/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h
:39:32: error: 'chrono' in namespace 'std' does not name a type
   void setStartTime(const std::chrono::time_point<std::chrono::steady_clock> time)                                                                                                          ^~~~~~
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h
:39:27: note: suggested alternative: 'conj'
   void setStartTime(const std::chrono::time_point<std::chrono::steady_clock> time)
                           ^~~                                                                                                                                                          conj
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h
:39:50: error: expected unqualified-id before '<' token
   void setStartTime(const std::chrono::time_point<std::chrono::steady_clock> time)                                                                                                                            ^
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h
:39:50: error: expected ')' before '<' token
   void setStartTime(const std::chrono::time_point<std::chrono::steady_clock> time)
                    ~                             ^                                                                                                                                                            )
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:39:40: error: expected ';' at end of member declaration
   void setStartTime(const std::chrono::time_point<std::chrono::steady_clock> time)
                                        ^~~~~~~~~~
                                                  ;                                                                                                          /home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:39:50: error: expected unqualified-id before '<' token
   void setStartTime(const std::chrono::time_point<std::chrono::steady_clock> time)                                                                                                                            ^                                                                                                          /home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h
:47:27: error: 'chrono' in namespace 'std' does not name a type

   std::chrono::steady_clock::duration totalTime() const;
   ^~~
   conj
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:101:8: error: 'chrono' in namespace 'std' does not name a type
   std::chrono::steady_clock::duration childrenTime() const;
        ^~~~~~
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:101:3: note: suggested alternative: 'conj'
   std::chrono::steady_clock::duration childrenTime() const;
   ^~~
   conj
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:113:8: error: 'chrono' in namespace 'std' does not name a type
   std::chrono::time_point<std::chrono::steady_clock> _start_time;
        ^~~~~~
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:113:3: note: suggested alternative: 'conj'
   std::chrono::time_point<std::chrono::steady_clock> _start_time;
   ^~~
   conj
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:116:8: error: 'chrono' in namespace 'std' does not name a type
   std::chrono::steady_clock::duration _total_time;
        ^~~~~~
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:116:3: note: suggested alternative: 'conj'
   std::chrono::steady_clock::duration _total_time;
   ^~~
   conj
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h: In constructor 'PerfNode::PerfNode(PerfID)':
/home/las_djorton/spack/opt/spack/linux-fedora33-skylake/gcc-8.4.0/moose-20-12-24-ht5jaewmwjovksfstgkyjaz2ibwtvcah/framework/build/header_symlinks/PerfNode.h:29:40: error: class 'PerfNode' does not have any field named '_total_time'
   PerfNode(const PerfID id) : _id(id), _total_time(0), _num_calls(0) {}

Any ideas on what I am doing wrong?

**Comment by aeslaughter** [comment]
@milljm Did you see this?

**Comment by loganharbour** [comment, marked as answer]
What do you get when you 

```
#include <chrono>
```

at the top of `PerfNode.h`?

It's possible that some includes are different within your stack, and we need to add an include there.

**Reply by loganharbour** [reply]
I see now that you said this:

> I played with editing the PerfNode.h file by adding in a "#include " statement, however, unless this is a bug, I do not think this should be the solution.

I assume that means that including chrono worked? It could be very well possible that you're missing that include from elsewhere in the framework which causes this, in which case the solution is indeed to include it.

I've created a PR to add this: #16681. Please confirm that the include resolves your issue and we'll get this merged.
